### DISCUSSION TITLE
Quantum doesn't seem compatible with Elixir 1.12

### DISCUSSION THREAD

**Discussion by hickscorp** [quantum-elixir/quantum-core#489] [Q&A]
When running Dialyzer over our project, we get the following errors:

```
lib/quantum.ex:269:unknown_function
Function GenStage.cast/2 does not exist.
________________________________________________________________________________
lib/quantum.ex:288:unknown_function
Function GenStage.cast/2 does not exist.
________________________________________________________________________________
lib/quantum.ex:294:unknown_function
Function GenStage.cast/2 does not exist.
________________________________________________________________________________
lib/quantum.ex:300:unknown_function
Function GenStage.cast/2 does not exist.
________________________________________________________________________________
lib/quantum.ex:306:unknown_function
Function GenStage.call/3 does not exist.
________________________________________________________________________________
lib/quantum.ex:312:unknown_function
Function GenStage.cast/2 does not exist.
________________________________________________________________________________
lib/quantum.ex:317:unknown_function
Function GenStage.cast/2 does not exist.
________________________________________________________________________________
lib/quantum.ex:322:unknown_function
Function GenStage.call/3 does not exist.
```

**Comment by maennchen** [comment]
@hickscorp that seems not to have anything to do with the elixir version.

Do you have any compile warning?

Did you clear the _build directory, deps directory and the dialyzer cache? Does it still happen then?

**Comment by hickscorp** [comment]
Yes - `_build` directory was nuked, as well as `deps` to get things built from scratch.
I don't recall seeing compile warnings.

**Comment by maennchen** [comment]
@hickscorp This does not seem to be a quantum issue. Quantum defines a dependency to `gen_stage` and therefore expects it to be present.

I'm currently updating the CI to the newest version. As you can see, dialyzer passes with current Elixir / Erlang Versions: https://github.com/quantum-elixir/quantum-core/runs/3815811406?check_suite_focus=true

Therefore I conclude, that this is not a quantum bug and will convert this issue to a discussion instead.
